src/goto-instrument/generate_function_bodies.cpp

```diff
@@ -209,26 +209,24 @@ class havoc_generate_function_bodiest : public generate_function_bodiest
       instruction->source_location = function_symbol.location;
       return instruction;
     };
+
     const namespacet ns(symbol_table);
-    for(auto const &parameter : function.type.parameters())
+    for(auto const &parameter : function.parameter_identifiers)
     {
+      const symbolt &parameter_symbol = ns.lookup(parameter);
       if(
-        parameter.type().id() == ID_pointer &&
-        !parameter.type().subtype().get_bool(ID_C_constant) &&
+        parameter_symbol.type.id() == ID_pointer &&
+        !parameter_symbol.type.subtype().get_bool(ID_C_constant) &&
         std::regex_match(
-          id2string(parameter.get_base_name()), parameters_to_havoc))
+          id2string(parameter_symbol.base_name), parameters_to_havoc))
       {
         auto goto_instruction =
           add_instruction(goto_programt::make_incomplete_goto(equal_exprt(
-            symbol_exprt(parameter.get_identifier(), parameter.type()),
-            null_pointer_exprt(to_pointer_type(parameter.type())))));
-
-        const irep_idt base_name = parameter.get_base_name();
-        CHECK_RETURN(!base_name.empty());
```

Comment:
:question: Is it intentional to remove this check of the parameters base name?

Reply:
Yes, because `base_name` is not actually used thereafter...